src/select/select-component.js

```diff
@@ -350,8 +350,8 @@ class Select extends React.Component<PropsT, SelectStateT> {
       case 13: // enter
         event.preventDefault();
         event.stopPropagation();
-        if (!this.state.isOpen) {
-          this.setState({isOpen: true});
+        if (this.state.isOpen && !this.props.multi) {
```

Comment:
I think this is not the right logic. When we press "enter", if it's a single select we need to close the menu and if it's a multi select we need to keep the menu open. Originally we make the menu open in any cases after "enter" and I don't know why. 

Besides, If we don't do this modification, the test (https://github.com/uber/baseweb/pull/3631#discussion_r470065126) (https://github.com/uber/baseweb/blob/7196b43a94bd22f2a1d8a519845cad5db861cc0e/src/select/__tests__/select.e2e.js#L311) above cannot pass.

Reply:
I don't really understand why the e2e would fail. Can you identify how the keydown listener change broke this test? Ideally we would not make any behavior changes in this diff

Reply:
Yes, it's quite strange here. I just found when we listen to `document` the menu will close automatically after "enter" and when we listen to a specific dom node the menu will stay open. I cannot figure out why.